controllers/jar.controllers.js

```diff
@@ -0,0 +1,493 @@
+    return next(error);
+  }
+
+  res.status(200).json({
+    basketData: [...jars],
+    paginationData: { maxPageAmount: Math.ceil(totalJarCount / onePageLimit) },
+  });
+}
+
+async function getPrivateJars(req, res, next) {
+  const { id } = req.user;
+  const { page, order } = req.query;
+
+  const bottomIndex = (page - 1) * onePageLimit;
+
+  const ownerJarCount = await Jar.countDocuments({
+    ownerId: ObjectId(id),
+    isPublic: false,
+  });
+  const coownerJarCount = await Participants.find({ user: ObjectId(id) })
+    .populate('basket')
+    .where('basket.isPublic')
+    .equals(false)
+    .countDocuments();
+  const totalJarCount = ownerJarCount + coownerJarCount;
+
+  let jars;
+  try {
+    jars = await Jar.aggregate([
+      { $match: { ownerId: ObjectId(id), isPublic: false } },
+      {
+        $unionWith: {
+          coll: 'participants',
+          pipeline: [
+            { $match: { user: ObjectId(id) } },
+            {
+              $lookup: {
+                from: 'baskets',
+                localField: 'basket',
+                foreignField: '_id',
+                as: 'basket',
+              },
+            },
+            { $replaceWith: { $first: '$basket' } },
+            { $match: { isPublic: false } },
+          ],
+        },
+      },
+      { $sort: getOrderArgs(order) },
+      { $skip: bottomIndex },
+      { $limit: onePageLimit },
+      {
+        $lookup: {
+          from: 'users',
+          localField: 'ownerId',
+          foreignField: '_id',
+          pipeline: [
+            { $project: { email: 0, password: 0, firstName: 0, lastName: 0 } },
+          ],
+          as: 'user',
+        },
+      },
+    ]);
+  } catch (error) {
+    return next(error);
+  }
+
+  res.status(200).json({
+    basketData: [...jars],
+    paginationData: { maxPageAmount: Math.ceil(totalJarCount / onePageLimit) },
+  });
+}
+
+async function createBasket(req, res, next) {
+  const errors = validationResult(req);
+
+  if (!errors.isEmpty()) {
+    return next(new HttpError('Invalid or not all inputs passed.', 422));
+  }
+
+  let stripeId;
+
+  try {
+    stripeId = await createCustomer({
+      email: '',
+      firstName: req.body.basketName,
+      lastName: 'Basket',
+    });
+  } catch (err) {
+    return next(
+      new HttpError(
+        `Error when creating a basket appeared. Message: ${err.message}`,
+        500,
+      ),
+    );
+  }
+
+  let basket;
+  try {
+    basket = await Jar.create({
+      ownerId: req.user._id,
+      name: req.body.basketName,
+      description: req.body.description,
+      goal: req.body.moneyGoal,
+      value: 0,
+      expirationDate: req.body.expirationDate,
+      isPublic: req.body.isPublic,
+      creationDate: +new Date(),
+      image: req.body.photoTag,
+      stripeId,
+    });
+  } catch (error) {
+    return next(
+      new HttpError(
+        `Error when creating a basket appeared. Message: ${error.message}`,
+        500,
+      ),
+    );
+  }
+
```

Comment:
Consider adding logger for cloud service
```suggestion
logger.info(jar created, {userId: id, jarId: basket._id})
```